### PR TITLE
disallow Timedelta ops with unitless timedelta64

### LINKED ISSUE #19388: Timedelta +/- timedelta64[no-unit] assumes ns
A test failure that only showed up on appveyor: https://ci.appveyor.com/project/pandas-dev/pandas/build/1.0.9563/job/b47phinjg410eiap

```
pd.Timedelta(hours=2) - np.timedelta64('NaT')

>   __sub__ = _binary_op_method_timedeltalike(lambda x, y: x - y, '__sub__')
E   TypeError: Cannot cast ufunc subtract input from dtype('<m8') to dtype('<m8[ns]') with casting rule 'same_kind'
```

On OSX this goes through without a hitch:

```
>>> pd.Timedelta(hours=2) - np.timedelta64(2)
numpy.timedelta64(7199999999998,'ns')
```

In this case the appveyor is probably right to raise instead of assuming nanos.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #19388
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -1237,6 +1249,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
...
+    value : np.timedelta64 or ndarray[timedelta64]
+
+    Raises
+    ------
+    ValueError
+    """
+    # GH#19388
+    if value.dtype == 'm8':
+        # i.e. has no unit
+        if util.is_array(value) or (value.view('i8') != NPY_NAT):
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: I think you can just do a direct check here

```
In [1]: np.timedelta64('NaT')
Out[1]: numpy.timedelta64('NaT')

In [2]: np.timedelta64('NaT') is np.timedelta64(1, 'ns')
Out[2]: False

In [3]: np.timedelta64('NaT') is np.array([1], dtype='m8[ns]')
Out[3]: False
```

IOW *only*  if it IS ``np.timedelta64('NaT')`` will it be  ok

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 2]
```diff
@@ -614,3 +614,32 @@ def test_rdivmod_invalid(self):
 
         with pytest.raises(TypeError):
             divmod(np.array([22, 24]), td)
+
+
+def test_require_td64_unit():
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: should be in test_constructor

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 3]
```diff
@@ -1237,6 +1249,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
         return res
 
 
+cdef _require_td64_has_unit(value):
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: type value (object) you can also add inline

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 4]
```diff
@@ -1237,6 +1249,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
...
+    """
+    Require that a timedelta64 scalar have a unit specified,
+    i.e. 'timedelta64[ns]' is OK but 'timedelta64' is not.
+
+    Parameters
+    ----------
+    value : np.timedelta64 or ndarray[timedelta64]
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: is this *always* true?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 4 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx]: Not quite; good catch.

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx]: Doesn't work like `pd.NaT`:

```
>>> np.timedelta64('NaT') is np.timedelta64('NaT')
False
```

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 5]
```diff
@@ -1270,6 +1282,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
...
+    ----------
+    value : object
+        assumed to have a `dtype` attribute
+
+    Raises
+    ------
+    ValueError
+    """
+    if value.dtype == 'm8':
+        # i.e. has no unit
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: pls make this more concise (e.g. a single if)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 5 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx]: The exception message could be shortened, but everything else is needed.  Combining the two ifs into one line would hurt clarity.

Comment (jreback) [reply pandas/_libs/tslibs/timedeltas.pyx]: you can just use ``missing.is_null_datetimelike`` instead of repeating code here. pls use a single if.

**Review by jreback — CHANGES_REQUESTED** [review]

**Pull request closed without merging**